src/app/clusters/microwave-oven-control-server/microwave-oven-control-server.cpp

```diff
@@ -0,0 +1,330 @@
+/**
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ *
+ */
+
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app/InteractionModelEngine.h>
+#include <app/clusters/microwave-oven-control-server/microwave-oven-control-server.h>
+#include <app/reporting/reporting.h>
+#include <app/util/attribute-storage.h>
+#include <app/util/error-mapping.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::OperationalState;
+using namespace chip::app::Clusters::MicrowaveOvenControl;
+using namespace chip::app::Clusters::MicrowaveOvenControl::Attributes;
+using Status = Protocols::InteractionModel::Status;
+
+namespace {
+
+constexpr size_t kMicrowaveOvenControlInstanceTableSize =
+    EMBER_AF_MICROWAVE_OVEN_CONTROL_CLUSTER_SERVER_ENDPOINT_COUNT + CHIP_DEVICE_CONFIG_DYNAMIC_ENDPOINT_COUNT;
+
+static_assert(kMicrowaveOvenControlInstanceTableSize <= kEmberInvalidEndpointIndex,
+              "Microwave Oven Control Instance table size error");
+
+OperationalState::Instance * gOPInstanceTable[kMicrowaveOvenControlInstanceTableSize] = { nullptr };
+
+OperationalState::Instance * gOperationalStateInstance = nullptr;
+
+} // anonymous namespace
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace MicrowaveOvenControl {
+
+Instance::Instance(Delegate * aDelegate, EndpointId aEndpointId, ClusterId aClusterId) :
+    CommandHandlerInterface(MakeOptional(aEndpointId), aClusterId), AttributeAccessInterface(MakeOptional(aEndpointId), aClusterId),
+    mDelegate(aDelegate), mEndpointId(aEndpointId), mClusterId(aClusterId)
+{
+    mDelegate->SetInstance(this);
+}
+
+Instance::~Instance()
+{
+    gOperationalStateInstance = nullptr;
```

Comment:
This doesn't make sense to me.  Why is shutting down a specific instance of the cluster affecting the gOperationalStateInstance global?

Reply:
If there are multi-endpoints in the device, the last MicrowaveOvenControl instance is shutiing down, this global `gOperationalStateInstance` should be set to NULL.

It may not be good to write this way because `gOPInstanceTable` has not delete yet if all the MicrowaveOvenControl are shutdown, but the code here will be changed to not use the global.

Reply:
Why is there a global gOperationalStateInstance at all?  There shouldn't be.